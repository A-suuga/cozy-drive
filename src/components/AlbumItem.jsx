import styles from '../styles/albumsList'

import React, { Component } from 'react'
// import classNames from 'classnames'
import { Link, withRouter } from 'react-router'
import { translate } from '../lib/I18n'

import { getPhotoLink } from '../actions/photos'

export class AlbumItem extends Component {
  constructor (props) {
    super(props)

    this.state = {
      isLoading: true
    }

    if (props.album && props.album.photosIds.length) {
      getPhotoLink(props.album.photosIds[0])
        .then(link => this.setState({
          url: link,
          isLoading: false
        }))
    } else {
      this.setState({
        url: '',
        isLoading: false
      })
    }
  }

  render () {
    const { t, album, router } = this.props
    const { isLoading, url } = this.state
    const parentPath = router.location.pathname
    return (
      !isLoading &&
        <div className={styles['pho-album']}>
          <Link
            to={`${parentPath}/${album._id}`}
            className={styles['pho-album-link']}
          >
            <img
              className={styles['pho-album-photo-item']}
              src={url}
            />
            <h2 className={styles['pho-album-title']}>{album.name}</h2>
            <h4 className={styles['pho-album-description']}>
              {t('Albums.album_item_description',
                {smart_count: album.photosIds.length})
              }
            </h4>
          </Link>
        </div>
    )
  }
}

export default translate()(withRouter(AlbumItem))
