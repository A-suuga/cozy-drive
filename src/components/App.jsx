import styles from '../styles/app'

import React from 'react'
import { translate } from '../lib/I18n'
import classNames from 'classnames'

import Sidebar from './Sidebar'

const App = ({ t, children }) => (
  <div className={classNames(styles['pho-wrapper'], styles['coz-sticky'])}>

    <Sidebar />

    <main className={styles['pho-content']}>
      { children }
    </main>
  </div>
)

export default translate()(App)
