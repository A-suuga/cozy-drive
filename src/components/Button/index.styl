@require 'cozy-ui'

@media (max-width: (1023/basefont)rem)
    .dri-btn--more
        display inline-flex

@media (max-width: (768/basefont)rem)
    .dri-btn--more
        border        0
        margin-right  0
        height        em(48px, 14px)
        opacity       .5