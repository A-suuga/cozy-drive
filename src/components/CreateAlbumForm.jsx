import styles from '../styles/createAlbumForm'

import React, { Component } from 'react'
import { translate } from '../lib/I18n'

import classNames from 'classNames'

class CreateAlbumForm extends Component {
  constructor () {
    super()
    this.reset()
  }

  reset () {
    this.setState({
      name: '',
      isSubmitDisabled: true,
      isBusy: false,
      hasError: false
    })
  }

  onSubmit (event, callback) {
    event.preventDefault()
    this.setState({ isBusy: true })
    return callback(this.state.name)
      .then(
        () => this.reset(),
        () => this.setState({
          hasError: true,
          isBusy: false
        })
      )
  }

  onNameChange (event) {
    const name = event.target.value.trim()
    this.setState({
      name: name,
      isSubmitDisabled: !name.length,
      hasError: false
    })
  }

  render ({ t, onSubmitNewAlbum }) {
    return (
      <form onSubmit={(event) => this.onSubmit(event, onSubmitNewAlbum)}>
        <label className={styles['coz-create-album-label']}>
          {t('Albums.create.inline_form.create_label')}
        </label>
        <div className={styles['coz-inline-form']}>
          <input
            className={classNames(styles['coz-input-text'], this.state.hasError && styles['error'])}
            type='text'
            name='album-name'
            onInput={(event) => this.onNameChange(event)}
            disabled={this.state.isBusy}
            placeholder={t('Albums.create.inline_form.placeholder')}
            value={this.state.name}
            />
          <button
            className={classNames('coz-btn', 'coz-btn--regular', styles['coz-btn'])}
            disabled={this.state.isSubmitDisabled || this.state.isBusy}
            aria-busy={this.state.isBusy}
            >
            {t('Albums.create.inline_form.create_button')}
          </button>
        </div>
      </form>
    )
  }
}

export default translate()(CreateAlbumForm)
