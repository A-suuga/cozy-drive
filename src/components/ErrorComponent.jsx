import styles from '../styles/errorComponent'

import React from 'react'
import { translate } from '../lib/I18n'

export const ErrorComponent = ({ t, errorType }) => {
  return (
    <div>
      {errorType === 'albums' &&
        <div className={styles['pho-error']}>
          <h2>{ t('Error.albums_title') }</h2>
          <button
            role='button'
            className='coz-btn coz-btn--regular'
            onClick={() => window.location.reload()}
          >
            {t('Error.refresh')}
          </button>
        </div>
      }
    </div>
  )
}

export default translate()(ErrorComponent)
