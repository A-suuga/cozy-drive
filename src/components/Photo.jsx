/* global cozy */
import styles from '../styles/photoList'

import React, { Component } from 'react'
import classNames from 'classnames'
import { Link, withRouter } from 'react-router'

import { getPhotoLink } from '../actions/photos'

const getStyleFromBox = box => {
  let style = {}
  if (box) {
    if (box.width) {
      style.width = `${box.width}px`
    }
    if (box.height) {
      style.height = `${box.height}px`
    }
  }
  return style
}

export class Photo extends Component {
  constructor (props) {
    super(props)
    this.state = {
      isImageLoading: true,
      url: `${cozy.client._url}${props.photo.links.small}`,
      fallback: null
    }
    this.handleImageLoaded = this.handleImageLoaded.bind(this)
    this.handleImageError = this.handleImageError.bind(this)
  }

  handleImageLoaded () {
    this.setState({ isImageLoading: false })
  }

  handleImageError () {
    if (this.state.fallback && this.img.src === this.state.fallback) return
    // extreme fallback
    getPhotoLink(this.props.photo._id)
      .then(url => {
        this.img.src = url
        this.setState({ fallback: url })
      })
  }

  render () {
    const { photo, box, selected = false, onToggle, router } = this.props
    const { loading, url, isImageLoading } = this.state
    const parentPath = router.location.pathname
    return (
      <div
        className={classNames(
          styles['pho-photo'],
          { [styles['pho-photo--selected']]: selected }
        )}
        style={getStyleFromBox(box)}
      >
        { !loading &&
          <div>
            <span
              className={styles['pho-photo-select']}
              data-input='checkbox'
              onClick={e => {
                e.stopImmediatePropagation()
                onToggle(photo._id, selected)
              }}>
              <input
                type='checkbox'
                checked={selected}
               />
              <label />
            </span>
            <Link to={`${parentPath}/${photo._id}`}>
              <img
                ref={img => { this.img = img }}
                className={styles['pho-photo-item']}
                onLoad={this.handleImageLoaded}
                onError={this.handleImageError}
                style={Object.assign(
                  getStyleFromBox(box),
                  isImageLoading ? {display: 'none'} : {})
                }
                alt={photo.name}
                src={url}
              />
            </Link>
          </div>
        }
      </div>
    )
  }
}

export default withRouter(Photo)
