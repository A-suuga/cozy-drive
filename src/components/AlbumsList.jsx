import styles from '../styles/albumsList'

import React from 'react'

import Empty from '../components/Empty'
import AlbumItem from '../components/AlbumItem'

export const AlbumsList = ({ t, f, albums, onServerError }) => {
  if (albums.length === 0) {
    return <Empty emptyType='albums' />
  } else {
    return (
      <div className={styles['pho-album-list']}>
        {albums.map((a) => <AlbumItem album={a} key={a._id} onServerError={onServerError} />)}
      </div>
    )
  }
}

export default AlbumsList
