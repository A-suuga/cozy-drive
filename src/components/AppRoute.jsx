import React from 'react'
import { Route, Redirect } from 'react-router'

import Timeline from '../containers/Timeline'
import AlbumsView from '../containers/AlbumsView'
import AlbumPhotos from '../containers/AlbumPhotos'
import Viewer from '../containers/Viewer'

import App from './App'
export const ComingSoon = () => (<p style='margin-left: 2em'>Coming soon!</p>)

const AppRoute = (
  <Route component={App}>
    <Redirect from='/' to='photos' />
    <Route path='photos' component={Timeline}>
      <Route path=':photoId' component={Viewer} />
    </Route>
    <Route path='albums' component={AlbumsView}>
      <Route path=':albumId' component={AlbumPhotos}>
        <Route path=':photoId' component={Viewer} />
      </Route>
    </Route>
  </Route>
)

export default AppRoute
