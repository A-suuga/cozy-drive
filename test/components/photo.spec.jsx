'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import { Photo } from '../../src/components/Photo'

const photoObject = {
  _id: '33dda00f0eec15bc3b3c59a615001ac8',
  created_at: '0001-01-01T00:00:00Z',
  name: 'MonImage.jpg',
  size: '150000',
  updated_at: '0001-01-01T00:00:00Z'
}

const routerObjectMock = {
  location: {
    pathname: '/mock'
  }
}

describe('Photo component', () => {
  it('should render correctly an image according a photo object and the router path', () => {
    const component = shallow(
      <Photo photo={photoObject} router={routerObjectMock} />
    )

    component.setState({
      loading: false,
      url: `http://cozy.local:8080/files/download/33dda00f0eec15bc3b3c59a615001ac8`
    })

    expect(component.node).toMatchSnapshot()
  })

  it('should handle onClick event', () => {
    const component = shallow(
      <Photo photo={photoObject} router={routerObjectMock}
        onToggle={() => {}} />
    )
    component.setState({
      loading: false,
      url: `http://cozy.local:8080/files/download/33dda00f0eec15bc3b3c59a615001ac8`
    })
    component.find('[data-input="checkbox"]').simulate('click', {stopImmediatePropagation: () => {}})
    expect(component.node).toMatchSnapshot()
  })
})
