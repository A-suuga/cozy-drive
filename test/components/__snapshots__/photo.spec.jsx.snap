exports[`Photo component should render correctly an image according a photo object and the router path 1`] = `
<Link
  onlyActiveOnIndex={false}
  style={Object {}}
  to="/mock/33dda00f0eec15bc3b3c59a615001ac8">
  <img
    className="pho-photo-item"
    src="http://cozy.local:8080/files/download/33dda00f0eec15bc3b3c59a615001ac8" />
</Link>
`;
